.pre-commit-config.yaml

```diff
@@ -5,3 +5,90 @@ repos:
     - id: black
       language_version: python3.6
 
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.780
+    hooks:
+    - id: mypy
+      exclude:
+          (?x)^(
+          docs/conf.py |
+          src/gt4py/analysis/infos.py |
+          src/gt4py/analysis/passes.py |
+          src/gt4py/analysis/transformer.py |
+          src/gt4py/backend/base_gt_backend.py |
+          src/gt4py/backend/debug_backend.py |
+          src/gt4py/backend/gt_cpu_backend.py |
+          src/gt4py/backend/gt_cuda_backend.py |
+          src/gt4py/backend/numpy_backend.py |
+          src/gt4py/backend/pyext_builder.py |
+          src/gt4py/backend/python_generator.py |
+          src/gt4py/frontend/gtscript_frontend.py |
+          src/gt4py/ir/nodes.py |
+          src/gt4py/ir/utils.py |
+          src/gt4py/stencil_object.py |
+          src/gt4py/utils/meta.py |
+          tests/_disabled/test_iir/test_validation.py |
+          tests/test_integration/stencil_definitions.py |
+          tests/test_integration/test_code_generation.py |
+          tests/test_unittest/test_call_interface.py |
+          tests/test_unittest/test_gtscript_frontend.py |
+          )$
+
+-   repo: http://gitlab.com/PyCQA/flake8
+    rev: 3.8.2
+    hooks:
+    - id: flake8
+      language_version: python3.6
```

Comment:
does this cause flake8 to only check for compatibility with py3.6 syntax? then we should maybe check all supported versions. (includes up to 3.8 right now)

Reply:
Not precisely, It makes sure that the python interpreter used to run flake8 is python3.6. However the pinned flake8 version supports up to 3.8.

We should probably set `default_language_version: 3.6` instead of setting for each hook, just so all hooks run in the same interpreter for consistency (and possibly less overhead at hook install time).

Reply:
The purpose of this whole thing is just so your local pre-commit runs exactly the same thing as the one on the CI. For flake8 it might not make a difference at all.

Reply:
Also it causes you to have to `pre-commit install-hooks` with a python3.6 available in the path. This ensures nobody develops on 3.8 and commits 3.8+ syntax without explicitly dropping support for <3.8 for example.

Reply:
It's a bit unrelated, but I think we should also update the `black` config on `pyproject.toml` to add python 3.8 to the `target_version` array.